lib/index.js

```diff
@@ -149,6 +149,44 @@ module.exports = function modelBase (bookshelf, params) {
     },
 
     /**
+      * Select a model based on data with ID and insert if not found
+      * @param {Object} data
+      * @param {Object} [options] Options for model.fetch and model.save
+      * @param {Object} [options.defaults] Defaults to apply to a create
+      * @param {String} id The model's ID
+      * @return {Promise(bookshelf.Model)} single Model
+      */
+    findOrCreateById: function (data, options, id) {
```

Comment:
to reduce the amount of code you have to write/test, you could just re-use the `.findOrCreatebyProperty` method here:

```js
findOrCreateById: function (data, options, id) {
  return this.findOrCreateByProperty(data, options, { [this.prototype.idAttribute]: id })
}
```